_includes/js/files.md

```diff
@@ -113,20 +113,17 @@ Parse.Cloud.httpRequest({ url: profilePhoto.url() }).then(function(response) {
 
 ## Deleting Files
 
-You can delete files that are referenced by objects using the [REST API]({{ site.baseUrl }}/rest/guide/#deleting-files). You will need to provide the master key in order to be allowed to delete a file.
-
-If your files are not referenced by any object in your app, it is not possible to delete them through the REST API.
-
-Alternatively, starting with Parse Server 4.2.0, you can delete files using cloud code.
+You can delete files that are referenced by objects using the `destroy` method. The master key is required to delete a file.
 
 ```javascript
-Parse.Cloud.beforeDelete('Profile', async (req) => {
-  const profile = req.object;
-  const profilePhoto = profile.get("photoFile");
-  await profilePhoto.destroy({ useMasterKey: true })
-});
+const profilePhoto = profile.get("photoFile");
+await profilePhoto.destroy({ useMasterKey: true });
 ```
 
+* Parse Server <4.2.0
+
+The `destroy` method is available since Parse Server 4.2.0, for lower versions use the [REST API]({{ site.baseUrl }}/rest/guide/#deleting-files) to delete a file.
+
```

Comment:
> I think it's unnecessary to mention the 4.2.0 requirement twice here.

Yea, that seems redundant. How about:
"~~The `destroy` method is available since Parse Server 4.2.0, for lower versions~~ Use the REST API to delete a file."

> I suggest removing the bullet point. 

My thinking was that the docs should always reflect the newest version. Any notes regarding previous versions are becoming negligible and should therefore be below, marked with their own bullet point, ordered by relevance descending. The layout would be:

< DOC FOR NEWEST VERSION, high relevance >

* Parse Server <4.2.0
< DOC FOR VERSION <4.2, medium relevance >

* Parse Server <2.2.0
< DOC FOR VERSION <2.2, low relevance >

Reply:
> * Parse Server <4.2.0
>   ~The `destroy` method is available since Parse Server 4.2.0, for lower versions~ Use the REST API to delete a file.

Wording wise that works

> 
> My thinking was that the docs should always reflect the newest version. Any notes regarding previous versions are becoming negligible and should therefore be below, marked with their own headline or bullet point, ordered by relevance descending. The layout would be:
> 
> < DOC FOR NEWEST VERSION, high relevance >
> 
> * Parse Server <4.2.0
>   < DOC FOR VERSION <4.2, medium relevance >
> * Parse Server <2.2.0
>   < DOC FOR VERSION <2.2, low relevance >
> 
> What do you think?

Ahh, I didn't realise you were using the bullet point as a form of heading. I agree with the hierarchy.

I don't think the layout for the older versions works as the line(s) below aren't indented. I would suggest either having the whole doc on the bullet point or using an actual heading like `###`.

Could you also add `Available only on Parse Server Cloud Code starting 4.2.0` under neither the main `Deleting Files` heading?

Reply:
> I don't think the layout for the older versions works as the line(s) below aren't indented. I would suggest either having the whole doc on the bullet point or using an actual heading like ###.

Right, that makes sense, I'll change that.

> Could you also add Available only on Parse Server Cloud Code starting 4.2.0 under neither the main Deleting Files heading?

When I look into the docs, I often stumble upon a "Available since parse server 2.x", and it never fails to irritate me. I think visual real estate should ideally contain only info that is useful, but these notes are not useful for maybe 99% of readers, because they use a version >2.x.

So instead of saying:

> ### Amazing feature
> ##### Available since Parse Server 4.2.
> Do this...


I would say:

> ### Amazing feature
> Do this...
> ##### Parse Server <4.2
> This feature is not available.

because it would be the same schema as:

> ### Amazing feature
> Do this...
> ##### Parse Server <4.2
> Do that instead...

This way the notes are used has headlines and denote their own doc block without interfering with the section that documents the status quo.

Reply:
I find this contradictory, you are saying that a 1 line note at the top irritates you and is not useful but also suggesting documenting potentially multiple older versions below?

Notes about features available on 2.x should probably be removed but personally I like the notes about recent releases at the top as it hopefully helps to prevent people wasting time trying to use features which don't exist on their version and helps to showcase the improvements being made across the platform.

Reply:
I think we think the same :)

What I mean is that a line at the top "available since..." is unnecessary for most readers and therefore should be avoided because it is placed before the information the reader is actually looking for. 

A line at the bottom "on lower version use workaround..." is not as disturbing since it is placed below the information most readers are looking for.

Now, whether it should be mentioned that a feature is not available or that there is a workaround for lower versions is a different question. I think it should, and there may be better forms of denoting the availability of features.

Reply:
Sorry for the lack of reply!

I'm not sure I agree with your assumptions about 'most readers' but I really don't have the head space to discuss this further at the moment.

I'll merge the PR as it's definitely as improvement regardless!